catalogs/catalog-jdbc-doris/src/main/java/com/datastrato/gravitino/catalog/doris/operation/DorisTablePartitionOperations.java

```diff
@@ -0,0 +1,313 @@
+        ImmutableList.Builder<Partition> partitions = ImmutableList.builder();
+        while (result.next()) {
+          partitions.add(fromDorisPartition(result, partitionInfo, columnTypes));
+        }
+        return partitions.build().stream().toArray(Partition[]::new);
+      }
+    } catch (SQLException e) {
+      throw exceptionConverter.toGravitinoException(e);
+    }
+  }
+
+  @Override
+  public Partition getPartition(String partitionName) throws NoSuchPartitionException {
+    try (Connection connection = getConnection(databaseName)) {
+      Transform partitionInfo = getPartitionInfo(connection);
+      Map<String, Type> columnTypes = getColumnType(connection);
+      String showPartitionsSql =
+          String.format(
+              "SHOW PARTITIONS FROM `%s` WHERE PartitionName = \"%s\"", tableName, partitionName);
+      try (Statement statement = connection.createStatement();
+          ResultSet result = statement.executeQuery(showPartitionsSql)) {
+        while (result.next()) {
+          return fromDorisPartition(result, partitionInfo, columnTypes);
+        }
+      }
+    } catch (SQLException e) {
+      throw exceptionConverter.toGravitinoException(e);
+    }
+    throw new NoSuchPartitionException("Partition %s does not exist", partitionName);
+  }
+
+  @Override
+  public Partition addPartition(Partition partition) throws PartitionAlreadyExistsException {
+    try (Connection connection = getConnection(databaseName)) {
+      Transform partitionInfo = getPartitionInfo(connection);
+
+      String addPartitionSqlFormat = "ALTER TABLE `%s` ADD PARTITION `%s` VALUES %s";
+      String addPartitionSql;
+      Partition added;
+
+      if (partition instanceof RangePartition) {
+        Preconditions.checkArgument(
+            partitionInfo instanceof Transforms.RangeTransform,
+            "Table %s is partitioned by list, but trying to add a range partition",
+            tableName);
+
+        RangePartition rangePartition = (RangePartition) partition;
+        Literal<?> upper = rangePartition.upper();
+        Literal<?> lower = rangePartition.lower();
+        String partitionValues;
+        if (Literals.NULL.equals(upper) && Literals.NULL.equals(lower)) {
+          partitionValues = "LESS THAN MAXVALUE";
+        } else if (Literals.NULL.equals(lower)) {
+          partitionValues = "LESS THAN (\"" + upper.value() + "\")";
+        } else if (Literals.NULL.equals(upper)) {
+          partitionValues = "[(\"" + lower.value() + "\"), (MAXVALUE))";
+        } else {
+          partitionValues = "[(\"" + lower.value() + "\"), (\"" + upper.value() + "\"))";
+        }
+        addPartitionSql =
+            String.format(addPartitionSqlFormat, tableName, rangePartition.name(), partitionValues);
+
+        // The partition properties actually cannot be passed into Doris, we just return an empty
+        // map instead.
+        added = Partitions.range(rangePartition.name(), upper, lower, Collections.emptyMap());
```

Comment:
Can you extract a method for these codes?

Reply:
Fixed.